### ISSUE TITLE
AttributeError: module 'google.protobuf.descriptor' has no attribute '_internal_create_key'

### ISSUE THREAD

**Issue by jcea** [SupraSummus/ipfs-api-mount#23]
Just trying recently released ipfs-api-mount 0.3.1, I get this error:

```
jcea@jcea:/tmp/a$ ipfs-api-mount
Traceback (most recent call last):
  File "/usr/local/bin/ipfs-api-mount", line 11, in <module>
    from ipfs_api_mount import IPFSMount, fuse_kwargs
  File "/usr/local/lib/python3.8/site-packages/ipfs_api_mount/__init__.py", line 9, in <module>
    from .ipfs import CachedIPFS, InvalidIPFSPathException
  File "/usr/local/lib/python3.8/site-packages/ipfs_api_mount/ipfs.py", line 7, in <module>
    from . import unixfs_pb2
  File "/usr/local/lib/python3.8/site-packages/ipfs_api_mount/unixfs_pb2.py", line 21, in <module>
    create_key=_descriptor._internal_create_key,
AttributeError: module 'google.protobuf.descriptor' has no attribute '_internal_create_key'
```

I tried also inside a virtualenv, so it seems not related to my configuration.

**Comment by SupraSummus** [comment]
It's probably version mismatch between installed protobuf compiler and protobuf python package. Haven't checked thoroughly though.

I've loosened requirements for protobuf version in https://github.com/SupraSummus/ipfs-api-mount/commit/fe5fd47e8813052185f7d14eb687772ba8c07f30. Please try installing from current master.

**Comment by SupraSummus** [comment]
Newer version of protobuf (3.12) seems to work fine. `pip install --upgrade protobuf` should solve the issue without reinstalling ipfs-api-mount.

Side note: The problem seems a bit deeper. Compiled protobuf code is included in `.whl` distribution archive so there will be similar errors every time protobuf version on target system is incompatible with protobuf compiler used during packaging. Loosening protobuf version bounds won't help at all ;/ I have to figure out how to package compiled protobuf properly.

**Comment by jcea** [comment]
Two obvious approaches:

1. Pin the protobuf dependency to a particular version.
2. Compile the protobuf code when importing this library, if needed.

**Comment by jcea** [comment]
@SupraSummus , aparently current "ipfs-api-mount" requires an outdated "protobuf" version and it doesn't actually work with it:

```
jcea@jcea:/tmp$ sudo python3 -m pip install --upgrade protobuf 
[sudo] password for jcea: 
WARNING: The directory '/home/jcea/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting protobuf
  Downloading protobuf-3.12.2-cp38-cp38-manylinux1_x86_64.whl (1.3 MB)
     |████████████████████████████████| 1.3 MB 820 kB/s 
Requirement already satisfied, skipping upgrade: six>=1.9 in /usr/local/lib/python3.8/site-packages (from protobuf) (1.14.0)
Requirement already satisfied, skipping upgrade: setuptools in /usr/local/lib/python3.8/site-packages (from protobuf) (41.2.0)
ERROR: ipfs-api-mount 0.3.1 has requirement protobuf==3.6.*, but you'll have protobuf 3.12.2 which is incompatible.
Installing collected packages: protobuf
  Attempting uninstall: protobuf
    Found existing installation: protobuf 3.6.1
    Uninstalling protobuf-3.6.1:
      Successfully uninstalled protobuf-3.6.1
Successfully installed protobuf-3.12.2
```

Note the line ```ERROR: ipfs-api-mount 0.3.1 has requirement protobuf==3.6.*, but you'll have protobuf 3.12.2 which is incompatible```.

**Comment by SupraSummus** [comment]
@jcea afaik there is no easy way to compile protobuf when importing the library. The closest approach is to install from source package. This will compile protobuf during package installation. First approach is good - but I have to somehow enforce a protoc version used when packaging. Without this restriction the fix is temporary - next protobuf upgrade will silently break things again.

**Comment by jcea** [comment]
Issue seems to be solved in release 0.3.2. Thanks.

I guess that improving protobuf management should be a different issue. @SupraSummus , feel free to close this bug and open a new one.

Thanks

**Comment by SupraSummus** [comment]
#35 <- iproving protobuf management
